core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -26,6 +27,10 @@
     private StatementRewriter statementRewriter;
     private TimingCollector timingCollector;
 
+    private boolean returningGeneratedKeys;
+    private String[] generatedKeysColumnNames;
```

Comment:
Actually, now that I look at it, it's not necessary to clone internally since the getter/setter always make a defensive clone. The array is kept private and never mutated, so can be shared between SqlStatements instances.